test/test_formats.py

```diff
@@ -158,6 +157,16 @@ def test_ratings(self):
             loaded_metadata = save_and_load_metadata(self.filename, metadata)
             self.assertEqual(int(loaded_metadata['~rating']), rating, '~rating: %r != %r' % (loaded_metadata['~rating'], rating))
 
+    def test_guess_format(self):
+        if self.original:
+            fd, temp_file = mkstemp()
+            os.close(fd)
+            shutil.copy(self.original, temp_file)
+            audio = picard.formats.guess_format(temp_file)
+            audio_original = picard.formats.open(self.filename)
+            self.assertEqual(type(audio), type(audio_original))
+            self.addCleanup(os.unlink,temp_file)
```

Comment:
This is just the same as calling `os.unlink(temp_file)` directly because all the interesting things happen before the call to `addCleanup` - imagine `guess_format` or `open` raising an error, the execution will never reach `addCleanup`. `addCleanup` is also only available in Python 2.7.